aspnetcore/host-and-deploy/webconfig-transformation.md

```diff
@@ -0,0 +1,163 @@
+
+If no profile is passed, the default profile name is **FileSystem** and *web.FileSystem.config* is applied if the file is present in the app's content root.
+
+## Environment
+
+Environment transformations are run third, after [Configuration](#configuration) and [Profile](#profile) transforms.
+
+Include a *web.{ENVIRONMENT}.config* file for each configuration requiring a *web.config* transformation.
+
+In the following example, a environment-specific environment variable is set in *web.Production.config*:
+
+```xml
+<?xml version="1.0"?>
+<configuration xmlns:xdt="http://schemas.microsoft.com/XML-Document-Transform">
+  <location>
+    <system.webServer>
+      <aspNetCore>
+        <environmentVariables xdt:Transform="InsertIfMissing">
+          <environmentVariable  name="Environment_Specific" value="Environment_Specific_Value" xdt:Locator="Match(name)" xdt:Transform="InsertIfMissing" />
+        </environmentVariables>
+      </aspNetCore>
+    </system.webServer>
+  </location>
+</configuration>
+```
+
+The transform is applied when the environment is *Production*:
+
+```console
+dotnet publish --configuration Release /p:EnvironmentName=Production
+```
+
+The MSBuild property for the environment is `$(EnvironmentName)`.
+
+When publishing from Visual Studio and using a publish profile, see <xref:host-and-deploy/visual-studio-publish-profiles#set-the-environment>.
+
+The `ASPNETCORE_ENVIRONMENT` environment variable is automatically added to the *web.config* file when the environment name is specified.
+
+## Custom
+
+Custom transformations are run last, after [Configuration](#configuration), [Profile](#profile), and [Environment](#environment) transforms.
+
+Include a *{CUSTOM_NAME}.transform* file for each configuration requiring a *web.config* transformation.
+
+In the following example, a custom transform environment variable is set in *custom.transform*:
+
+```xml
+<?xml version="1.0"?>
+<configuration xmlns:xdt="http://schemas.microsoft.com/XML-Document-Transform">
+  <location>
+    <system.webServer>
+      <aspNetCore>
+        <environmentVariables xdt:Transform="InsertIfMissing">
+          <environmentVariable  name="Custom_Specific" value="Custom_Specific_Value" xdt:Locator="Match(name)" xdt:Transform="InsertIfMissing" />
+        </environmentVariables>
+      </aspNetCore>
+    </system.webServer>
+  </location>
+</configuration>
+```
+
+The transform is applied when the `CustomTransformFileName` property is passed to the [dotnet publish](/dotnet/core/tools/dotnet-publish) command:
+
+```console
+dotnet publish --configuration Release /p:CustomTransformFileName=custom.transform
+```
+
+The MSBuild property for the profile name is `$(CustomTransformFileName)`.
+
+## Prevent web.config transformation
+
+To prevent transformations of the *web.config* file, set the MSBuild property `$(IsWebConfigTransformDisabled)`:
+
+```console
+dotnet publish /p:IsWebConfigTransformDisabled=true
+```
```

Comment:
Link to xdt syntax reference in an additional resources section
The only one I know of is archived 
https://docs.microsoft.com/en-us/previous-versions/aspnet/dd465326(v=vs.110) 
If we really expect people to use this in production we should get the xdt syntax doc out of mothballs.